ladybug_geometry/polyskel.py

```diff
@@ -0,0 +1,694 @@
+	return a == b or (abs(a - b) <= max(abs(a), abs(b)) * 0.001)
+
+def _approximately_same(point_a, point_b):
+	return _approximately_equals(point_a.x, point_b.x) and \
+		   _approximately_equals(point_a.y, point_b.y)
+
+def _normalize_contour(contour):
+	"""
+	Consumes list of x,y coordinate tuples and returns list of Point2Ds.
+
+	Args:
+		contour: list of x,y tuples from contour.
+	Return:
+		 list of Point2Ds of contour.
+	"""
+	contour = [Point2D(float(x), float(y)) for (x,y) in contour]
+	normed_contour = []
+	for prev, point, next in _window(contour):
+		normed_prev = (point-prev).normalize()
+		normed_next = (next-point).normalize()
+		if not (point == next or normed_prev == normed_next):
+			normed_contour.append(point)
+
+	return normed_contour
+
+
+class _SplitEvent(
+	namedtuple(
+		"_SplitEvent",
+		"distance, "
+		"intersection_point, "
+		"vertex, "
+		"opposite_edge"
+		)
+	):
+
+	__slots__ = ()
+
+	def __str__(self):
+		return "{} Split event @ {} from {} to {}".format(
+			self.distance,
+			self.intersection_point,
+			self.vertex,
+			self.opposite_edge
+			)
+
+class _EdgeEvent(
+	namedtuple(
+		"_EdgeEvent",
+		"distance intersection_point vertex_a vertex_b"
+		)
+	):
+
+	__slots__ = ()
+
+	def __str__(self):
+		return "{} Edge event @ {} between {} and {}".format(
+			self.distance,
+			self.intersection_point,
+			self.vertex_a,
+			self.vertex_b
+			)
+
+_OriginalEdge = namedtuple("_OriginalEdge", "edge bisector_left, bisector_right")
+
+Subtree = namedtuple("Subtree", "source, height, sinks")
+
+def _side(point, line):
+	a = line.p.x
+	b = line.p.y
+
+class _LAVertex:
+	def __init__(self, point, edge_left, edge_right, direction_vectors=None):
+		self.point = point
+		self.edge_left = edge_left
+		self.edge_right = edge_right
+		self.prev = None
+		self.next = None
+		self.lav = None
+		# this should be handled better. Maybe membership in lav implies validity?
+		self._valid = True
+		edge_left._v = edge_left.v.normalize()
+		edge_right._v = edge_right.v.normalize()
+		creator_vectors = (edge_left.v * -1, edge_right.v)
+		if direction_vectors is None:
+			direction_vectors = creator_vectors
+
+		# The determinant of two 2d vectors equals the sign of their cross product
+		# If second vector is to left, then sign of det is pos, else neg
+		# So if cw polygon, convex angle will be neg (since second vector to right)
+		# In this case, since we flip first vector, concave will be neg
+		self._is_reflex = _determinant(*direction_vectors) < 0.0
+		self._bisector = Ray2D(
+			self.point,
+			operator.add(*creator_vectors) * (-1 if self.is_reflex else 1)
+			)
+		log.info("Created vertex %s", self.__repr__())
+		_debug.line((self.bisector.p.x, self.bisector.p.y,
+					 self.bisector.p.x + self.bisector.v.x * 100,
+					 self.bisector.p.y + self.bisector.v.y * 100), fill="blue")
+
+	@property
+	def bisector(self):
+		return self._bisector
+
+	@property
+	def is_reflex(self):
+		return self._is_reflex
```

Comment:
Sure. A vertex in this case is vertice in a list of vertices (specifically a double-connected circular list of other vertices). So it's essentially computing the angle between this vertex and angle with neighboring vertices.

Reply:
Ah, I think you might have just schooled me about my own understanding of the word "vertice". I see that it's actually distinct from "vertex" and the distinction is useful. Ok, we can keep using "vertice" where it's appropriate as long as we're consistent in our definition. And including what you just explained here in a docstring would be very helpful.

Reply:
Actually... I think you're correct, and I'm the one using vertex/vertice incorrectly: http://www.3drender.com/glossary/vertex.htm

Apparently vertice is not a word!?

Reply:
Good to know. I guess we'll continue to use vertex, then, but it would still be good to have a docstring explaining that the vertex this class represents is always a part of a larger list of vertices.